### ISSUE TITLE
Question: Has anyone succeeded in training img2text for image captioning problem?

### ISSUE THREAD

**Issue by anphu7492** [OpenNMT/OpenNMT-py#898]
Hi, I'm wondering if anyone can share experience on training an image captioning system using `ImageEncoder`. I've trained on MSCOCO dataset with default `ImageEncoder` architecture but the validation accuracy can never exceed 50% and the BLEU score is very low (around `8.xx`). I tried to modify `ImageEncoder` to use `ResNet152` architecture but the results were even worse.

**Comment by da03** [comment]
Hmm that's weird, this image encoder is adapted from vgg so shouldn't be too bad on image captioning problems. What's your training and validation perplexity (with original code)?

**Comment by anphu7492** [comment]
Hi @da03 , this is my training results after 100k steps:
```
Step 100000/100000; acc:  51.88; ppl:  7.45; xent: 2.01; lr: 0.00156;
```
and validation results: 
```
Validation perplexity: 19.1125
Validation accuracy: 45.3577
```
I'm wondering if 100k step training is enough. The validation accuracy didn't increase much (even decreased a little bit) and vice versa for perplexity.

**Comment by da03** [comment]
Hmm with this ppl seems that the model is not using image features (a language model might be able to achieve this ppl). How about trying to use Adam with learning_rate 0.001? And can you share me the training data such that I can take a look?

**Comment by anphu7492** [comment]
I'm trying to train with `lr=1` and `Adam` (previously trained with  default `lr=0.1`. I'll try `0.001` after that if my current configurations don't work. It will converge really slowly with `lr=0.001` I guess. Here is the training example that I have with only first 100 images (the file for the whole dataset is about 50GB). You can ignore the `src_feats` field in the data, I'm not using them. The file can be downloaded from here: https://www.dropbox.com/s/afy4g78yge0194o/coco_sm.train.pt?dl=0

**Comment by da03** [comment]
Oh for Adam learning rate should be smaller than SGD, usually we do 1e-3.

**Comment by da03** [comment]
Hmm weird, what's the size of the training set?

**Comment by anphu7492** [comment]
I'm training on COCO (2014) dataset, so the training set size is 82783 and validation set size is 40504

**Comment by da03** [comment]
That should be large enough, not sure what's the reason for the low generalization ability. Can you try using training set as validation set? (I suspect it's related to batchnorm)

**Comment by anphu7492** [comment]
Did you mean to use training set as both training and validation sets?

**Comment by anphu7492** [comment]
Ok, I'll try that next.

**Comment by anphu7492** [comment]
Using training set as validation set produced the following:
Train stats:
```
[2018-08-15 23:47:48,750 INFO] Step 65500/79000; acc:  77.81; ppl:  2.27; xent: 0.82; lr: 0.00025;   0/882 tok/s;    408 sec
```
Valid stats:
```
[2018-08-15 23:54:28,040 INFO] Validation perplexity: 1.588
[2018-08-15 23:54:28,040 INFO] Validation accuracy: 87.5149
```

**Comment by da03** [comment]
Hmm interesting, that means it's indeed overfitting. Does validation dataset have a different distribution or a different preprocessing from training set? Maybe data augmentation is needed (which is not implemented in this code), or use larger dropout?

**Comment by anphu7492** [comment]
Training and validation sets were created at the same time by the same processing script (`preprocess.py`). I'm not sure what the problem is, I doubt that changing dropout would help but maybe worth a try. I've seen people training on MSCOCO dataset and they succeeded without data augmentation :(

**Comment by vince62s** [comment]
closing, if you need just reopen it.
